startup.py

```diff
@@ -106,12 +106,47 @@ def prepare_launch(self, exec_path, args, file_to_open=None):
         # https://github.com/shotgunsoftware/tk-shotgun-launchpublish
         # https://github.com/shotgunsoftware/tk-config-default2/blob/master/env/publishedfile.yml#L27
         # https://github.com/shotgunsoftware/tk-config-default2/blob/master/env/includes/settings/tk-shotgun-launchpublish.yml
-
-
-        self.logger.debug("Launch environment: %s", pprint.pformat(required_env))
-        self.logger.debug("Launch arguments: %s", args)
-
-        return LaunchInformation(exec_path, args, required_env)
+        
+        #Get user project path with Qt File Dialog          
+        from sgtk.util.qt_importer import QtImporter
+        qt = QtImporter()
+        QFileDialog = qt.QtGui.QFileDialog
+        QMessageBox = qt.QtGui.QMessageBox
+        QApplication = qt.QtGui.QApplication
+        QErrorMessage = qt.QtGui.QErrorMessage
+        # options for either browse type
+        options = [
+            QFileDialog.DontResolveSymlinks,
+            QFileDialog.ShowDirsOnly,
+        ]
+        # browse folders specifics
+        caption = "Select a Unity Project Directory"
+        file_mode = QFileDialog.Directory
+        # create the dialog    
+        file_dialog = QFileDialog(
+            parent=QApplication.instance().activeWindow(), caption=caption
+        )
+        file_dialog.setLabelText(QFileDialog.Accept, "Select")
+        file_dialog.setLabelText(QFileDialog.Reject, "Cancel")
+        file_dialog.setFileMode(file_mode)
+        
+        # set the appropriate options
+        for option in options:
+            file_dialog.setOption(option)
+        # browse!
+        while file_dialog.exec_():
+                selected_file_path = file_dialog.selectedFiles()[0]
+                if os.path.exists(selected_file_path+"\Assets"):
```

Comment:
This will not work on Linux/MacOS.
You need to build paths using `os.path.join(selected_file_path, "Assets")`